src/sdk/pynni/nni/compression/torch/pruning/one_shot.py

```diff
@@ -141,6 +185,66 @@ def validate_config(self, model, config_list):
 
         schema.validate(config_list)
 
+    def _dependency_calc_mask(self, wrappers, channel_dsets, wrappers_idx=None):
+        """
+        calculate the masks for the conv layers in the same
+        channel dependecy set. All the layers passed in have
+        the same number of channels.
+
+        Parameters
+        ----------
+        wrappers : list
+            The list of the wrappers that in the same channel dependency
+            set.
+        wrappers_idx : list
+            The list of the indexes of wrapppers.
+        """
+        # The number of the groups for each conv layers
+        # Note that, this number may be different from its
+        # original number of groups of filters.
+        groups = [self.group_depen[_w.name] for _w in wrappers]
+        masks = self.masker.calc_mask(
+            wrappers, channel_dsets, groups, wrappers_idx=wrappers_idx)
+        if masks is not None:
+            # if masks is None, then the mask calculation fails.
+            # for example, in activation related maskers, we should
+            # pass enough batches of data to the model, so that the
+            # masks can be calculated successfully.
+            for _w in wrappers:
+                _w.if_calculated = True
+        return masks
+
+    def _dependency_update_mask(self):
+        """
+        In the original update_mask, the wraper of each layer will update its
+        mask own mask according to the sparsity specified in the config_list. However, in
+        the _dependency_update_mask, we may prune several layers at the same
+        time according the sparsities and the channel/group depedencies.
+        """
+        logger.info('Update the mask in a dependency-aware way')
+        name2wraper = {x.name: x for x in self.get_modules_wrapper()}
+        wraper2index = {x: i for i, x in enumerate(self.get_modules_wrapper())}
+        for wrapper in self.get_modules_wrapper():
+            if wrapper.if_calculated:
+                continue
+            # find all the conv layers that have channel dependecy with this layer
+            # and prune all these layers at the same time.
+            _names = [x for x in self.channel_depen[wrapper.name]]
+            logger.info('Pruning the dependent layers: %s', ','.join(_names))
+            _wrappers = [name2wraper[name]
+                         for name in _names if name in name2wraper]
```

Comment:
what if some `conv`s in a dependency group are not specified with sparsity?

Reply:
I think this again after the meeting, it's not a bug, the code already filter out the layers that are not pruned
```
for name in _names if name in name2wraper
```
if the name not in `name2wrapper` which means this layer is not pruned, then we won't call `name2wrapper[name]`.